{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fd09362b-ee61-495c-9a6c-7ddec6a76648",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: numpy in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 1)) (1.23.5)\n",
      "Collecting opencv-python\n",
      "  Downloading opencv_python-4.7.0.68-cp37-abi3-win_amd64.whl (38.2 MB)\n",
      "     --------------------------------------- 38.2/38.2 MB 10.7 MB/s eta 0:00:00\n",
      "Collecting loguru\n",
      "  Downloading loguru-0.6.0-py3-none-any.whl (58 kB)\n",
      "     ---------------------------------------- 58.3/58.3 kB ? eta 0:00:00\n",
      "Requirement already satisfied: scikit-image in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 4)) (0.19.3)\n",
      "Requirement already satisfied: scikit-learn in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 5)) (1.0.2)\n",
      "Requirement already satisfied: tqdm in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 6)) (4.64.1)\n",
      "Collecting torchvision>=0.10.0\n",
      "  Downloading torchvision-0.14.1-cp38-cp38-win_amd64.whl (1.1 MB)\n",
      "     ---------------------------------------- 1.1/1.1 MB 9.9 MB/s eta 0:00:00\n",
      "Requirement already satisfied: Pillow in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 8)) (9.3.0)\n",
      "Collecting thop\n",
      "  Downloading thop-0.1.1.post2209072238-py3-none-any.whl (15 kB)\n",
      "Collecting ninja\n",
      "  Downloading ninja-1.11.1-py2.py3-none-win_amd64.whl (313 kB)\n",
      "     -------------------------------------- 313.0/313.0 kB 9.8 MB/s eta 0:00:00\n",
      "Requirement already satisfied: tabulate in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 11)) (0.8.10)\n",
      "Requirement already satisfied: tensorboard in c:\\users\\futak\\appdata\\roaming\\python\\python38\\site-packages (from -r requirements.txt (line 12)) (2.10.0)\n",
      "Collecting lap\n",
      "  Downloading lap-0.4.0.tar.gz (1.5 MB)\n",
      "     ---------------------------------------- 1.5/1.5 MB 10.6 MB/s eta 0:00:00\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Collecting motmetrics\n",
      "  Downloading motmetrics-1.4.0-py3-none-any.whl (161 kB)\n",
      "     ------------------------------------- 161.5/161.5 kB 10.1 MB/s eta 0:00:00\n",
      "Collecting filterpy\n",
      "  Downloading filterpy-1.4.5.zip (177 kB)\n",
      "     ------------------------------------- 178.0/178.0 kB 10.5 MB/s eta 0:00:00\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Requirement already satisfied: h5py in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 16)) (3.7.0)\n",
      "Requirement already satisfied: matplotlib in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 17)) (3.6.2)\n",
      "Requirement already satisfied: scipy in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 18)) (1.9.3)\n",
      "Collecting prettytable\n",
      "  Downloading prettytable-3.6.0-py3-none-any.whl (27 kB)\n",
      "Collecting easydict\n",
      "  Downloading easydict-1.10.tar.gz (6.4 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Requirement already satisfied: pyyaml in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from -r requirements.txt (line 22)) (6.0)\n",
      "Collecting yacs\n",
      "  Downloading yacs-0.1.8-py3-none-any.whl (14 kB)\n",
      "Requirement already satisfied: termcolor in c:\\users\\futak\\appdata\\roaming\\python\\python38\\site-packages (from -r requirements.txt (line 24)) (2.0.1)\n",
      "Collecting gdown\n",
      "  Downloading gdown-4.6.0-py3-none-any.whl (14 kB)\n",
      "Collecting onnx==1.8.1\n",
      "  Downloading onnx-1.8.1-cp38-cp38-win_amd64.whl (6.9 MB)\n",
      "     ---------------------------------------- 6.9/6.9 MB 11.3 MB/s eta 0:00:00\n",
      "Collecting onnxruntime==1.8.0\n",
      "  Downloading onnxruntime-1.8.0-cp38-cp38-win_amd64.whl (4.7 MB)\n",
      "     ---------------------------------------- 4.7/4.7 MB 6.2 MB/s eta 0:00:00\n",
      "Collecting onnx-simplifier==0.3.5\n",
      "  Downloading onnx-simplifier-0.3.5.tar.gz (13 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Requirement already satisfied: typing-extensions>=3.6.2.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from onnx==1.8.1->-r requirements.txt (line 26)) (4.4.0)\n",
      "Requirement already satisfied: six in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from onnx==1.8.1->-r requirements.txt (line 26)) (1.16.0)\n",
      "Collecting protobuf\n",
      "  Downloading protobuf-4.21.12-cp38-cp38-win_amd64.whl (527 kB)\n",
      "     ------------------------------------- 527.0/527.0 kB 11.0 MB/s eta 0:00:00\n",
      "Requirement already satisfied: flatbuffers in c:\\users\\futak\\appdata\\roaming\\python\\python38\\site-packages (from onnxruntime==1.8.0->-r requirements.txt (line 27)) (2.0.7)\n",
      "Collecting onnxoptimizer>=0.2.5\n",
      "  Downloading onnxoptimizer-0.3.6-cp38-cp38-win_amd64.whl (327 kB)\n",
      "     ------------------------------------- 327.9/327.9 kB 10.2 MB/s eta 0:00:00\n",
      "Collecting win32-setctime>=1.0.0\n",
      "  Downloading win32_setctime-1.1.0-py3-none-any.whl (3.6 kB)\n",
      "Requirement already satisfied: colorama>=0.3.4 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from loguru->-r requirements.txt (line 3)) (0.4.6)\n",
      "Requirement already satisfied: imageio>=2.4.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from scikit-image->-r requirements.txt (line 4)) (2.19.3)\n",
      "Requirement already satisfied: tifffile>=2019.7.26 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from scikit-image->-r requirements.txt (line 4)) (2021.7.2)\n",
      "Requirement already satisfied: packaging>=20.0 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from scikit-image->-r requirements.txt (line 4)) (22.0)\n",
      "Requirement already satisfied: networkx>=2.2 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from scikit-image->-r requirements.txt (line 4)) (2.8.4)\n",
      "Requirement already satisfied: PyWavelets>=1.1.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from scikit-image->-r requirements.txt (line 4)) (1.4.1)\n",
      "Requirement already satisfied: joblib>=0.11 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from scikit-learn->-r requirements.txt (line 5)) (1.1.1)\n",
      "Requirement already satisfied: threadpoolctl>=2.0.0 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from scikit-learn->-r requirements.txt (line 5)) (2.2.0)\n",
      "Requirement already satisfied: requests in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from torchvision>=0.10.0->-r requirements.txt (line 7)) (2.28.1)\n",
      "Collecting torch==1.13.1\n",
      "  Downloading torch-1.13.1-cp38-cp38-win_amd64.whl (162.6 MB)\n",
      "     -------------------------------------- 162.6/162.6 MB 8.1 MB/s eta 0:00:00\n",
      "Requirement already satisfied: werkzeug>=1.0.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from tensorboard->-r requirements.txt (line 12)) (2.2.2)\n",
      "Requirement already satisfied: tensorboard-data-server<0.7.0,>=0.6.0 in c:\\users\\futak\\appdata\\roaming\\python\\python38\\site-packages (from tensorboard->-r requirements.txt (line 12)) (0.6.1)\n",
      "Requirement already satisfied: markdown>=2.6.8 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from tensorboard->-r requirements.txt (line 12)) (3.4.1)\n",
      "Collecting grpcio>=1.24.3\n",
      "  Downloading grpcio-1.51.1-cp38-cp38-win_amd64.whl (3.7 MB)\n",
      "     ---------------------------------------- 3.7/3.7 MB 11.4 MB/s eta 0:00:00\n",
      "Requirement already satisfied: absl-py>=0.4 in c:\\users\\futak\\appdata\\roaming\\python\\python38\\site-packages (from tensorboard->-r requirements.txt (line 12)) (1.2.0)\n",
      "Collecting protobuf\n",
      "  Downloading protobuf-3.19.6-cp38-cp38-win_amd64.whl (896 kB)\n",
      "     ------------------------------------- 896.1/896.1 kB 11.4 MB/s eta 0:00:00\n",
      "Requirement already satisfied: tensorboard-plugin-wit>=1.6.0 in c:\\users\\futak\\appdata\\roaming\\python\\python38\\site-packages (from tensorboard->-r requirements.txt (line 12)) (1.8.1)\n",
      "Collecting google-auth<3,>=1.6.3\n",
      "  Downloading google_auth-2.16.0-py2.py3-none-any.whl (177 kB)\n",
      "     ------------------------------------- 177.8/177.8 kB 11.2 MB/s eta 0:00:00\n",
      "Requirement already satisfied: setuptools>=41.0.0 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from tensorboard->-r requirements.txt (line 12)) (65.6.3)\n",
      "Requirement already satisfied: wheel>=0.26 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from tensorboard->-r requirements.txt (line 12)) (0.37.1)\n",
      "Requirement already satisfied: google-auth-oauthlib<0.5,>=0.4.1 in c:\\users\\futak\\appdata\\roaming\\python\\python38\\site-packages (from tensorboard->-r requirements.txt (line 12)) (0.4.6)\n",
      "Requirement already satisfied: pandas>=0.23.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from motmetrics->-r requirements.txt (line 14)) (1.5.2)\n",
      "Collecting xmltodict>=0.12.0\n",
      "  Downloading xmltodict-0.13.0-py2.py3-none-any.whl (10.0 kB)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from matplotlib->-r requirements.txt (line 17)) (1.4.4)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from matplotlib->-r requirements.txt (line 17)) (4.25.0)\n",
      "Requirement already satisfied: cycler>=0.10 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from matplotlib->-r requirements.txt (line 17)) (0.11.0)\n",
      "Requirement already satisfied: contourpy>=1.0.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from matplotlib->-r requirements.txt (line 17)) (1.0.5)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from matplotlib->-r requirements.txt (line 17)) (2.8.2)\n",
      "Requirement already satisfied: pyparsing>=2.2.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from matplotlib->-r requirements.txt (line 17)) (3.0.9)\n",
      "Requirement already satisfied: wcwidth in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from prettytable->-r requirements.txt (line 19)) (0.2.5)\n",
      "Requirement already satisfied: beautifulsoup4 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from gdown->-r requirements.txt (line 25)) (4.11.1)\n",
      "Requirement already satisfied: filelock in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from gdown->-r requirements.txt (line 25)) (3.9.0)\n",
      "Collecting rsa<5,>=3.1.4\n",
      "  Downloading rsa-4.9-py3-none-any.whl (34 kB)\n",
      "Collecting cachetools<6.0,>=2.0.0\n",
      "  Downloading cachetools-5.3.0-py3-none-any.whl (9.3 kB)\n",
      "Requirement already satisfied: pyasn1-modules>=0.2.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from google-auth<3,>=1.6.3->tensorboard->-r requirements.txt (line 12)) (0.2.8)\n",
      "Requirement already satisfied: requests-oauthlib>=0.7.0 in c:\\users\\futak\\appdata\\roaming\\python\\python38\\site-packages (from google-auth-oauthlib<0.5,>=0.4.1->tensorboard->-r requirements.txt (line 12)) (1.3.1)\n",
      "Requirement already satisfied: importlib-metadata>=4.4 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from markdown>=2.6.8->tensorboard->-r requirements.txt (line 12)) (4.11.3)\n",
      "Requirement already satisfied: pytz>=2020.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from pandas>=0.23.1->motmetrics->-r requirements.txt (line 14)) (2022.7)\n",
      "Requirement already satisfied: urllib3<1.27,>=1.21.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from requests->torchvision>=0.10.0->-r requirements.txt (line 7)) (1.26.14)\n",
      "Requirement already satisfied: certifi>=2017.4.17 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from requests->torchvision>=0.10.0->-r requirements.txt (line 7)) (2022.12.7)\n",
      "Requirement already satisfied: idna<4,>=2.5 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from requests->torchvision>=0.10.0->-r requirements.txt (line 7)) (3.4)\n",
      "Requirement already satisfied: charset-normalizer<3,>=2 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from requests->torchvision>=0.10.0->-r requirements.txt (line 7)) (2.0.4)\n",
      "Requirement already satisfied: MarkupSafe>=2.1.1 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from werkzeug>=1.0.1->tensorboard->-r requirements.txt (line 12)) (2.1.1)\n",
      "Requirement already satisfied: soupsieve>1.2 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from beautifulsoup4->gdown->-r requirements.txt (line 25)) (2.3.2.post1)\n",
      "Requirement already satisfied: PySocks!=1.5.7,>=1.5.6 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from requests->torchvision>=0.10.0->-r requirements.txt (line 7)) (1.7.1)\n",
      "Requirement already satisfied: zipp>=0.5 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from importlib-metadata>=4.4->markdown>=2.6.8->tensorboard->-r requirements.txt (line 12)) (3.11.0)\n",
      "Requirement already satisfied: pyasn1<0.5.0,>=0.4.6 in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (from pyasn1-modules>=0.2.1->google-auth<3,>=1.6.3->tensorboard->-r requirements.txt (line 12)) (0.4.8)\n",
      "Requirement already satisfied: oauthlib>=3.0.0 in c:\\users\\futak\\appdata\\roaming\\python\\python38\\site-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard->-r requirements.txt (line 12)) (3.2.1)\n",
      "Building wheels for collected packages: onnx-simplifier, lap, filterpy, easydict\n",
      "  Building wheel for onnx-simplifier (setup.py): started\n",
      "  Building wheel for onnx-simplifier (setup.py): finished with status 'done'\n",
      "  Created wheel for onnx-simplifier: filename=onnx_simplifier-0.3.5-py3-none-any.whl size=12862 sha256=6184eb856adcbb212d638e5e70ad520982da641ecfd25bde9e7288d72560432e\n",
      "  Stored in directory: c:\\users\\futak\\appdata\\local\\pip\\cache\\wheels\\fd\\30\\8e\\518f4064830c3014a870bbc6abd8f6e5203804c96b07e83e29\n",
      "  Building wheel for lap (setup.py): started\n",
      "  Building wheel for lap (setup.py): finished with status 'done'\n",
      "  Created wheel for lap: filename=lap-0.4.0-cp38-cp38-win_amd64.whl size=1468087 sha256=c8b298640c9ad62906f589b821ac00de2cc3df22ea30147cd7244624f677fa5e\n",
      "  Stored in directory: c:\\users\\futak\\appdata\\local\\pip\\cache\\wheels\\15\\ac\\7b\\b08f663530e7c2b2bf2bb9ffaea75f564294bdebab9fdc4751\n",
      "  Building wheel for filterpy (setup.py): started\n",
      "  Building wheel for filterpy (setup.py): finished with status 'done'\n",
      "  Created wheel for filterpy: filename=filterpy-1.4.5-py3-none-any.whl size=110458 sha256=522e01128dfdb497fab72f296bcc5a0e131bac621096b762c5754a28bfd2e693\n",
      "  Stored in directory: c:\\users\\futak\\appdata\\local\\pip\\cache\\wheels\\b1\\68\\a2\\6981e4f3250879aabf325c9302cabd7c4973d6dd81b3017e99\n",
      "  Building wheel for easydict (setup.py): started\n",
      "  Building wheel for easydict (setup.py): finished with status 'done'\n",
      "  Created wheel for easydict: filename=easydict-1.10-py3-none-any.whl size=6492 sha256=3bfa6988c96eb8561c1b52901c4630b0f47bb59d7bb156fd06ab55d994bcda2b\n",
      "  Stored in directory: c:\\users\\futak\\appdata\\local\\pip\\cache\\wheels\\7b\\d2\\a1\\c4e0795ef16ab3276fb4ef17d3df108a15654c70157cd69ed3\n",
      "Successfully built onnx-simplifier lap filterpy easydict\n",
      "Installing collected packages: ninja, lap, easydict, yacs, xmltodict, win32-setctime, torch, rsa, protobuf, prettytable, opencv-python, grpcio, cachetools, torchvision, thop, onnxruntime, onnx, loguru, google-auth, onnxoptimizer, motmetrics, gdown, filterpy, onnx-simplifier\n",
      "Successfully installed cachetools-5.3.0 easydict-1.10 filterpy-1.4.5 gdown-4.6.0 google-auth-2.16.0 grpcio-1.51.1 lap-0.4.0 loguru-0.6.0 motmetrics-1.4.0 ninja-1.11.1 onnx-1.8.1 onnx-simplifier-0.3.5 onnxoptimizer-0.3.6 onnxruntime-1.8.0 opencv-python-4.7.0.68 prettytable-3.6.0 protobuf-3.19.6 rsa-4.9 thop-0.1.1.post2209072238 torch-1.13.1 torchvision-0.14.1 win32-setctime-1.1.0 xmltodict-0.13.0 yacs-0.1.8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No CUDA runtime is found, using CUDA_HOME='C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v10.1'\n",
      "Traceback (most recent call last):\n",
      "  File \"setup.py\", line 51, in <module>\n",
      "    long_description = f.read()\n",
      "UnicodeDecodeError: 'cp932' codec can't decode byte 0x82 in position 3997: illegal multibyte sequence\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: cython in c:\\users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages (0.29.28)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ERROR: Invalid requirement: \"'git+https://github.com/cocodataset/cocoapi.git#subdirectory=PythonAPI'\"\n",
      "Hint: = is not a valid operator. Did you mean == ?\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting cython_bbox\n",
      "  Downloading cython_bbox-0.1.3.tar.gz (41 kB)\n",
      "     ---------------------------------------- 41.3/41.3 kB ? eta 0:00:00\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Building wheels for collected packages: cython_bbox\n",
      "  Building wheel for cython_bbox (setup.py): started\n",
      "  Building wheel for cython_bbox (setup.py): finished with status 'error'\n",
      "  Running setup.py clean for cython_bbox\n",
      "Failed to build cython_bbox\n",
      "Installing collected packages: cython_bbox\n",
      "  Running setup.py install for cython_bbox: started\n",
      "  Running setup.py install for cython_bbox: finished with status 'error'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  python setup.py bdist_wheel did not run successfully.\n",
      "  exit code: 1\n",
      "  \n",
      "  [11 lines of output]\n",
      "  running bdist_wheel\n",
      "  running build\n",
      "  running build_ext\n",
      "  building 'cython_bbox' extension\n",
      "  creating build\n",
      "  creating build\\temp.win-amd64-cpython-38\n",
      "  creating build\\temp.win-amd64-cpython-38\\Release\n",
      "  creating build\\temp.win-amd64-cpython-38\\Release\\src\n",
      "  \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.28.29333\\bin\\HostX86\\x64\\cl.exe\" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\\Users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages\\numpy\\core\\include -IC:\\Users\\futak\\anaconda3\\envs\\multicam\\include -IC:\\Users\\futak\\anaconda3\\envs\\multicam\\Include \"-IC:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.28.29333\\include\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt\" /Tcsrc/cython_bbox.c /Fobuild\\temp.win-amd64-cpython-38\\Release\\src/cython_bbox.obj -Wno-cpp\n",
      "  cl : コマンド ライン error D8021 : 数値型引数 '/Wno-cpp' は無効です。\n",
      "  error: command 'C:\\\\Program Files (x86)\\\\Microsoft Visual Studio\\\\2019\\\\BuildTools\\\\VC\\\\Tools\\\\MSVC\\\\14.28.29333\\\\bin\\\\HostX86\\\\x64\\\\cl.exe' failed with exit code 2\n",
      "  [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "  ERROR: Failed building wheel for cython_bbox\n",
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  Running setup.py install for cython_bbox did not run successfully.\n",
      "  exit code: 1\n",
      "  \n",
      "  [13 lines of output]\n",
      "  running install\n",
      "  C:\\Users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages\\setuptools\\command\\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.\n",
      "    warnings.warn(\n",
      "  running build\n",
      "  running build_ext\n",
      "  building 'cython_bbox' extension\n",
      "  creating build\n",
      "  creating build\\temp.win-amd64-cpython-38\n",
      "  creating build\\temp.win-amd64-cpython-38\\Release\n",
      "  creating build\\temp.win-amd64-cpython-38\\Release\\src\n",
      "  \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.28.29333\\bin\\HostX86\\x64\\cl.exe\" /c /nologo /O2 /W3 /GL /DNDEBUG /MD -IC:\\Users\\futak\\anaconda3\\envs\\multicam\\lib\\site-packages\\numpy\\core\\include -IC:\\Users\\futak\\anaconda3\\envs\\multicam\\include -IC:\\Users\\futak\\anaconda3\\envs\\multicam\\Include \"-IC:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.28.29333\\include\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\ucrt\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\shared\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\um\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\winrt\" \"-IC:\\Program Files (x86)\\Windows Kits\\10\\include\\10.0.18362.0\\cppwinrt\" /Tcsrc/cython_bbox.c /Fobuild\\temp.win-amd64-cpython-38\\Release\\src/cython_bbox.obj -Wno-cpp\n",
      "  cl : コマンド ライン error D8021 : 数値型引数 '/Wno-cpp' は無効です。\n",
      "  error: command 'C:\\\\Program Files (x86)\\\\Microsoft Visual Studio\\\\2019\\\\BuildTools\\\\VC\\\\Tools\\\\MSVC\\\\14.28.29333\\\\bin\\\\HostX86\\\\x64\\\\cl.exe' failed with exit code 2\n",
      "  [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "error: legacy-install-failure\n",
      "\n",
      "Encountered error while trying to install package.\n",
      "\n",
      "cython_bbox\n",
      "\n",
      "note: This is an issue with the package mentioned above, not pip.\n",
      "hint: See above for output from the failure.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting faiss-cpu\n",
      "  Downloading faiss_cpu-1.7.3-cp38-cp38-win_amd64.whl (10.4 MB)\n",
      "     ---------------------------------------- 10.4/10.4 MB 9.3 MB/s eta 0:00:00\n",
      "Installing collected packages: faiss-cpu\n",
      "Successfully installed faiss-cpu-1.7.3\n",
      "Collecting faiss-gpu"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  WARNING: Generating metadata for package faiss-gpu produced metadata for project name faiss-cpu. Fix your #egg=faiss-gpu fragments.\n",
      "  WARNING: Generating metadata for package faiss-gpu produced metadata for project name faiss-cpu. Fix your #egg=faiss-gpu fragments.\n",
      "  WARNING: Generating metadata for package faiss-gpu produced metadata for project name faiss-cpu. Fix your #egg=faiss-gpu fragments.\n",
      "  WARNING: Generating metadata for package faiss-gpu produced metadata for project name faiss-cpu. Fix your #egg=faiss-gpu fragments.\n",
      "  WARNING: Generating metadata for package faiss-gpu produced metadata for project name faiss-cpu. Fix your #egg=faiss-gpu fragments.\n",
      "  WARNING: Generating metadata for package faiss-gpu produced metadata for project name faiss-cpu. Fix your #egg=faiss-gpu fragments.\n",
      "  error: subprocess-exited-with-error\n",
      "  \n",
      "  python setup.py egg_info did not run successfully.\n",
      "  exit code: 1\n",
      "  \n",
      "  [7 lines of output]\n",
      "  running egg_info\n",
      "  creating C:\\Users\\futak\\AppData\\Local\\Temp\\pip-pip-egg-info-8cek8flx\\faiss_cpu.egg-info\n",
      "  writing C:\\Users\\futak\\AppData\\Local\\Temp\\pip-pip-egg-info-8cek8flx\\faiss_cpu.egg-info\\PKG-INFO\n",
      "  writing dependency_links to C:\\Users\\futak\\AppData\\Local\\Temp\\pip-pip-egg-info-8cek8flx\\faiss_cpu.egg-info\\dependency_links.txt\n",
      "  writing top-level names to C:\\Users\\futak\\AppData\\Local\\Temp\\pip-pip-egg-info-8cek8flx\\faiss_cpu.egg-info\\top_level.txt\n",
      "  writing manifest file 'C:\\Users\\futak\\AppData\\Local\\Temp\\pip-pip-egg-info-8cek8flx\\faiss_cpu.egg-info\\SOURCES.txt'\n",
      "  error: package directory 'C:\\Users\\futak\\AppData\\Local\\Temp\\pip-install-j3_eqsmc\\faiss-gpu_3e8b63400b6f48fa9a1c9ab8134097ac\\faiss\\faiss\\python' does not exist\n",
      "  [end of output]\n",
      "  \n",
      "  note: This error originates from a subprocess, and is likely not a problem with pip.\n",
      "error: metadata-generation-failed\n",
      "\n",
      "Encountered error while generating package metadata.\n",
      "\n",
      "See above for output.\n",
      "\n",
      "note: This is an issue with the package mentioned above, not pip.\n",
      "hint: See above for details.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "  Downloading faiss-gpu-1.7.1.post2.tar.gz (40 kB)\n",
      "     ---------------------------------------- 40.3/40.3 kB ? eta 0:00:00\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Discarding https://files.pythonhosted.org/packages/17/76/47d0cc8161f4bf988583a2839bb1e56baf09d6b80cfa472b9eba4d5f543b/faiss-gpu-1.7.1.post2.tar.gz (from https://pypi.org/simple/faiss-gpu/): Requested faiss-cpu from https://files.pythonhosted.org/packages/17/76/47d0cc8161f4bf988583a2839bb1e56baf09d6b80cfa472b9eba4d5f543b/faiss-gpu-1.7.1.post2.tar.gz has inconsistent name: expected 'faiss-gpu', but metadata has 'faiss-cpu'\n",
      "  Downloading faiss-gpu-1.7.1.post1.tar.gz (41 kB)\n",
      "     ---------------------------------------- 41.1/41.1 kB ? eta 0:00:00\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Discarding https://files.pythonhosted.org/packages/39/8d/b62bc92c8dd4b2a99d4a06b8804280f6445748b6d698eabb037e111080c7/faiss-gpu-1.7.1.post1.tar.gz (from https://pypi.org/simple/faiss-gpu/): Requested faiss-cpu from https://files.pythonhosted.org/packages/39/8d/b62bc92c8dd4b2a99d4a06b8804280f6445748b6d698eabb037e111080c7/faiss-gpu-1.7.1.post1.tar.gz has inconsistent name: expected 'faiss-gpu', but metadata has 'faiss-cpu'\n",
      "  Downloading faiss-gpu-1.7.1.tar.gz (40 kB)\n",
      "     ---------------------------------------- 40.3/40.3 kB ? eta 0:00:00\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Discarding https://files.pythonhosted.org/packages/51/85/7a7490dbecaea9272953b88e236a45fe8c47571a069bc28b352f0b224ea3/faiss-gpu-1.7.1.tar.gz (from https://pypi.org/simple/faiss-gpu/): Requested faiss-cpu from https://files.pythonhosted.org/packages/51/85/7a7490dbecaea9272953b88e236a45fe8c47571a069bc28b352f0b224ea3/faiss-gpu-1.7.1.tar.gz has inconsistent name: expected 'faiss-gpu', but metadata has 'faiss-cpu'\n",
      "  Downloading faiss-gpu-1.7.0.tar.gz (34 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Discarding https://files.pythonhosted.org/packages/63/15/289ecf5d23f209c4c6f2f5f4db1d2b4a2be22d1fc49619354363e9367c19/faiss-gpu-1.7.0.tar.gz (from https://pypi.org/simple/faiss-gpu/): Requested faiss-cpu from https://files.pythonhosted.org/packages/63/15/289ecf5d23f209c4c6f2f5f4db1d2b4a2be22d1fc49619354363e9367c19/faiss-gpu-1.7.0.tar.gz has inconsistent name: expected 'faiss-gpu', but metadata has 'faiss-cpu'\n",
      "  Downloading faiss-gpu-1.6.5.tar.gz (28 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Discarding https://files.pythonhosted.org/packages/9c/27/3477c856ea3d678619c33ae72f89ede4fbe08e9c5ba3b89a5feb3d9938b0/faiss-gpu-1.6.5.tar.gz (from https://pypi.org/simple/faiss-gpu/): Requested faiss-cpu from https://files.pythonhosted.org/packages/9c/27/3477c856ea3d678619c33ae72f89ede4fbe08e9c5ba3b89a5feb3d9938b0/faiss-gpu-1.6.5.tar.gz has inconsistent name: expected 'faiss-gpu', but metadata has 'faiss-cpu'\n",
      "  Downloading faiss-gpu-1.6.4.post2.tar.gz (25 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'done'\n",
      "Discarding https://files.pythonhosted.org/packages/7d/00/b3aaad408a44e4f5d87ebfcd75d0b14eeaed9fe9bc7a9f5e185ff1d503d6/faiss-gpu-1.6.4.post2.tar.gz (from https://pypi.org/simple/faiss-gpu/): Requested faiss-cpu from https://files.pythonhosted.org/packages/7d/00/b3aaad408a44e4f5d87ebfcd75d0b14eeaed9fe9bc7a9f5e185ff1d503d6/faiss-gpu-1.6.4.post2.tar.gz has inconsistent name: expected 'faiss-gpu', but metadata has 'faiss-cpu'\n",
      "  Downloading faiss-gpu-1.6.4.tar.gz (3.4 kB)\n",
      "  Preparing metadata (setup.py): started\n",
      "  Preparing metadata (setup.py): finished with status 'error'\n"
     ]
    }
   ],
   "source": [
    "!pip install -r requirements.txt\n",
    "!python setup.py develop\n",
    "\n",
    "!pip install cython\n",
    "!pip install 'git+https://github.com/cocodataset/cocoapi.git#subdirectory=PythonAPI'\n",
    "# Cython-bbox\n",
    "!pip install cython_bbox\n",
    "\n",
    "# faiss cpu / gpu\n",
    "!pip install faiss-cpu\n",
    "!pip install faiss-gpu"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b7793a43-c88e-4487-bdb4-bf7077c6c10e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
